### PR TITLE
Adding AbstractBlocklyFragment and demo SimpleFragment

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Also clarifying comments about the save path overrides.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-android/647)

### CODE REVIEW

File: blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
+package com.google.blockly.android;
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java]: copyright

File: blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+import com.google.blockly.model.Mutator;
+import com.google.blockly.utils.BlockLoadingException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+
+public abstract class AbstractBlocklyFragment extends Fragment {
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java]: Missing class doc

File: blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+     * {@link BlocklyActivityHelper#saveWorkspaceToAppDir(String)} using
+     * {@link #getWorkspaceSavePath()}.
+     */
+    public void onSaveWorkspace() {
+        mBlocklyActivityHelper.saveWorkspaceToAppDirSafely(getWorkspaceSavePath());
+    }
+
+    /**
+     * Called when the user clicks the load action.  Default implementation delegates handling to
+     * {@link BlocklyActivityHelper#loadWorkspaceFromAppDir(String)}.
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java]: add "using {@link #getWorkspaceSavePath()}" to match comment above.

**Comment by AnmAtAnm** [comment]
Review status: 0 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 1 at r1](https://reviewable.io:443/reviews/google/blockly-android/647#-KsoB7Y6geJaixkE8MYs-r1-1:-KsoiX7zxhOcFYaDcf3T:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/0fdf16933669f8ce6373fd99d5bfead45e4e204b/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L1)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

copyright
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 32 at r1](https://reviewable.io:443/reviews/google/blockly-android/647#-KsoB7Y6geJaixkE8MYs-r1-32:-KsoiX_tGMexRgxablM6:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/0fdf16933669f8ce6373fd99d5bfead45e4e204b/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L32)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

Missing class doc
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 83 at r1](https://reviewable.io:443/reviews/google/blockly-android/647#-KsoB7Y6geJaixkE8MYs-r1-83:-Ksov6-XHL_xWob3iRA8:bmncj5u) ([raw file](https://github.com/google/blockly-android/blob/0fdf16933669f8ce6373fd99d5bfead45e4e204b/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L83)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

add "using {@link #getWorkspaceSavePath()}" to match comment above.
</blockquote></details>

Done. Here and in AbstractBlocklyActivity.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/647)*

**Comment by RoboErikG** [comment]
Review status: 0 of 14 files reviewed at latest revision, 6 unresolved discussions.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 66 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-KsovwQqcOqD3wrTI0Jn:-KsovwQqcOqD3wrTI0Jo:b5ccqvm) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L66)):*
> ```Java
>  * bundle.  If no workspace is loaded, it defers to {@link #onLoadInitialWorkspace}.
>  * <p/>
>  * Configure the workspace by providing definitions for {@link #getBlockDefinitionsJsonPaths()},
> ```

nit: replace comma with ' and'

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 250 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-Ksox759cmUFXu9z1rwk:-Ksox759cmUFXu9z1rwl:bmsuifq) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L250)):*
> ```Java
>     }
> 
>     public boolean onOptionsItemSelected(MenuItem item) {
> ```

missing method javadoc

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 256 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-KsoxFaC_WJ4YzwyFOE4:-KsoxFaC_WJ4YzwyFOE5:b80uzmu) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L256)):*
> ```Java
>         int id = item.getItemId();
> 
>         if (id == R.id.action_save) {
> ```

Should this be a switch instead of an if/else?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/647)*

**Comment by AnmAtAnm** [comment]
Review status: 0 of 14 files reviewed at latest revision, 6 unresolved discussions.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 66 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-KsovwQqcOqD3wrTI0Jn:-Ksp2-KBsNy2URhvwhDG:blv876c) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L66)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

nit: replace comma with ' and'
</blockquote></details>

Fixed here and in AbstractBlocklyActivity

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 250 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-Ksox759cmUFXu9z1rwk:-Ksp4FgqaDzqt8YHcLR8:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L250)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

missing method javadoc
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java, line 256 at r2](https://reviewable.io:443/reviews/google/blockly-android/647#-KsoxFaC_WJ4YzwyFOE4:-Ksp4ZgJE7YP3riTkxWF:b1xcihe) ([raw file](https://github.com/google/blockly-android/blob/6f6cdaf228f939bc6c96ff265578948ea1fe27a0/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyFragment.java#L256)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

Should this be a switch instead of an if/else?
</blockquote></details>

"Resource IDs cannot be used in switch statement in Android library modules."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/647)*

**Pull request merged**